parse/src/test/java/com/parse/AndroidKeyStoreProvider.kt

```diff
@@ -0,0 +1,140 @@
+
+        override fun engineGetCertificateAlias(cert: Certificate?): String? = wrapped.getCertificateAlias(cert)
+
+        override fun engineGetKey(alias: String?, password: CharArray?): Key? = (storedKeys[alias] as? KeyStore.SecretKeyEntry)?.secretKey
+
+        override fun engineGetEntry(p0: String, p1: KeyStore.ProtectionParameter?): KeyStore.Entry? = storedKeys[p0]
+
+        override fun engineSetEntry(p0: String, p1: KeyStore.Entry, p2: KeyStore.ProtectionParameter?) {
+            storedKeys[p0] = p1
+        }
+
+        override fun engineLoad(p0: KeyStore.LoadStoreParameter?) = wrapped.load(p0)
+
+        override fun engineStore(p0: KeyStore.LoadStoreParameter?) = wrapped.store(p0)
+
+        override fun engineEntryInstanceOf(p0: String?, p1: Class<out KeyStore.Entry>?) = wrapped.entryInstanceOf(p0, p1)
+
+        companion object {
+            private val wrapped = KeyStore.getInstance("BKS", "BC")
+            internal val storedKeys = mutableMapOf<String, KeyStore.Entry>()
+        }
+    }
+
+    class AesKeyGenerator : KeyGeneratorSpi() {
+        private val wrapped = KeyGenerator.getInstance("AES", "BC")
+        private var lastSpec: AlgorithmParameterSpec? = null
+
+        override fun engineInit(random: SecureRandom?) = wrapped.init(random)
+
+        override fun engineInit(params: AlgorithmParameterSpec?, random: SecureRandom?) = wrapped.init(random).also {
+            lastSpec = params
+        }
+
+        override fun engineInit(keysize: Int, random: SecureRandom?) = wrapped.init(keysize, random)
+
+        override fun engineGenerateKey(): SecretKey = wrapped.generateKey().also {
+            AndroidKeyStore.storedKeys[lastSpec!!.keystoreAlias] = KeyStore.SecretKeyEntry(it)
+        }
+    }
+
+    class HmacSHA256KeyGenerator : KeyGeneratorSpi() {
+        private val wrapped = KeyGenerator.getInstance("HmacSHA256", "BC")
+        private var lastSpec: AlgorithmParameterSpec? = null
+
+        override fun engineInit(random: SecureRandom?) = wrapped.init(random)
+        override fun engineInit(params: AlgorithmParameterSpec?, random: SecureRandom?) = wrapped.init(random).also {
+            lastSpec = params
+        }
+
+        override fun engineInit(keysize: Int, random: SecureRandom?) = Unit
+        override fun engineGenerateKey(): SecretKey = wrapped.generateKey().also {
+            AndroidKeyStore.storedKeys[lastSpec!!.keystoreAlias] = KeyStore.SecretKeyEntry(it)
+        }
+    }
+
+    class RsaKeyPairGenerator : KeyPairGeneratorSpi() {
+        private val wrapped = KeyPairGenerator.getInstance("RSA", "BC")
+
+        private var lastSpec: AlgorithmParameterSpec? = null
+
+        override fun generateKeyPair(): KeyPair = wrapped.generateKeyPair().also { keyPair ->
+            null
+//            AndroidKeyStore.storedKeys[lastSpec!!.keystoreAlias] = KeyStore.PrivateKeyEntry(keyPair.private, arrayOf(keyPair.toCertificate()))
```

Comment:
Let's not have commented code. 

If this is a TODO or a FIXME or something important for reference later on, please consider adding a descriptive comment instead of deleting it.